utils/utils.py

```diff
@@ -302,35 +309,47 @@ def get_hf_trained_models(organization: str = "sb3") -> Dict[str, Tuple[str, str
     :return: Dict representing the trained agents
     """
     api = HfApi()
-    models = api.list_models(author=organization)
-    regex = re.compile(r"^(?P<algo>[a-z_0-9]+)-(?P<env_id>[a-zA-Z0-9]+-v[0-9]+)$")
+    models = api.list_models(author=organization, cardData=True)
+
     trained_models = {}
     for model in models:
-        # Remove organization
-        repo_id = model.modelId.split(f"{organization}/")[1]
-        result = regex.match(repo_id)
-        # Skip demo repo that does not fit the pattern
-        if result is not None:
-            algo, env_id = result.group("algo"), result.group("env_id")
-            trained_models[f"{algo}-{env_id}"] = (algo, env_id)
+
+        # Try to extract algorithm and environment id from model card
+        try:
+            env_id = model.cardData["model-index"][0]["results"][0]["dataset"]["name"]
+            algo = model.cardData["model-index"][0]["name"].lower()
+        except KeyError:
+            continue  # skip model if name env id or algo name could not be found
+        except IndexError:
+            continue  # skip model if name env id or algo name could not be found
+
+        env_name = EnvironmentName(env_id)
+        model_name = ModelName(algo, env_name)
+
+        # check if there is a model file in the repo
+        if not any(f.rfilename == model_name.filename for f in api.model_info(model.modelId).siblings):
```

Comment:
we make one additional api call per model :/ ?
is there no way to filter things before? are "-" allowed in env id?

Reply:
We don't necessarily need this extra check. The alternative would be to assume, that all models of the organization, that have a dataset name in their results section are valid sb3-zoo models.

"-" should be allowed since it is part of every env id? I don't get that question here.

Reply:
> "-" should be allowed since it is part of every env id? I don't get that question here.

I meant that the only dashes that we should get is for the version and for converting slashes, no?
if so, we should be able to filter models more easily

Reply:
TBH I am afraid of adding more assumptions about what characters are or are not in an env id to the code.

If we want to filter properly and more efficiently, we should rather add an sb3-zoo flag to the model metadata when uploading to the hub. But pleeease can this go into a different PR? :-) I really want to finally close this one.